[Doc] Better backport/branch label documentation (#23530)
DataDog/datadog-agent: docs/dev/contributing.md

- `backport/<branch-name>`: Add this label to automatically create a PR against the `<branch-name>` branch with your backported changes. The backport PR creation is triggered:
  - When a PR with the label is merged
  - When an already-merged PR gets the label

>   In case of conflicts, the bot will prompt a list of instructions to follow ([example](https://github.com/DataDog/datadog-agent/pull/23316#issuecomment-1973207164)) to manually backport your PR.

Reviewer: ```suggestion
  If there is a conflict, the bot prompts you with a list of instructions to follow ([example](https://github.com/DataDog/datadog-agent/pull/23316#issuecomment-1973207164)) to manually backport your PR.
```
